lib/TcpConnection.js

```diff
@@ -141,3 +157,18 @@ function buildCliSourceStream () {
   inputStream.pause()
   return inputStream
 }
+
+/**
+ * @param {TcpConnectionOptions} settings
+ * @returns {tls.ConnectionOptions}
+ */
+function getConnectionOptions (settings) {
+  /** @type {tls.ConnectionOptions} */
+  const options = settings.unixsocket ? { path: settings.unixsocket } : { host: settings.address, port: settings.port }
+
+  if (settings.secure !== false && settings.unauth !== false) {
```

Comment:
This check would pass if unauth is set to undefined. Please avoid that as it can create security issues.